test/run_portability_test.py

```diff
@@ -0,0 +1,199 @@
+    if len(wdl_paths) > 1:
+        errors.append("Multiple candidate test WDLs found: {}".format(wdl_paths))
+
+    try:
+        test_wdl_string = open(wdl_paths[0]).read()
+    except IOError:
+        test_wdl_string = None
+        errors.append("Test WDL {} could not be read".format(wdl_paths[0]))
+
+    # Test inputs
+    try:
+        inputs_json_path = os.path.join(test_dir, TEST_DIR_LAYOUT["inputs"])
+        inputs_json_string = open(inputs_json_path).read()
+    except IOError:
+        inputs_json_string = None
+        errors.append("Inputs JSON {} could not be read".format(inputs_json_path))
+
+    # Dependencies
+
+    # First try to load the dependency JSON itself
+    try:
+        dependencies_json_path = os.path.join(test_dir, TEST_DIR_LAYOUT["dependencies"])
+        dependencies_json_string = open(dependencies_json_path).read()
+    except IOError:
+        dependencies_json_string = None
+        errors.append("Dependencies JSON {} could not be read".format(dependencies_json_path))
+
+    # Now iterate over the dependencies and load the files.
+    loaded_dependencies = []
+    dependencies_dict = json.loads(dependencies_json_string)
+
+    for key, value in dependencies_dict.items():
+        try:
+            loaded_dependencies.append({
+                "name": key,
+                "code": open(value).read()
+            })
+        except IOError:
+            errors.append("Count not read dependency {}".format(value))
+
+    if errors:
+        for error in errors:
+            sys.stderr.write(error + "\n")
+        raise TestDefinitionError(errors)
+
+    return {
+        "workflow_descriptor": test_wdl_string,
+        "workflow_params": inputs_json_string,
+        "workflow_dependencies": loaded_dependencies
+        }
+
+def verify_environment_variables():
+    """Check that required environment variables are defined."""
+
+    errors = []
+
+    for key, value in ENV_VARIABLES.items():
+        if not value:
+            errors.append("Environment variable {} is undefined".format(key.upper()))
+    if errors:
+        for error in errors:
+            sys.stderr.write(error + "\n")
+        raise TestEnvironmentError(errors)
+
+def submit_portability_test(test_inputs):
+    """Submit the portability to the service and return the test's id."""
+
+    service_headers = json.loads(ENV_VARIABLES["portability_service_headers"])
+    test_endpoint = ENV_VARIABLES["portability_service_url"] + "/portability_tests"
+    response = requests.post(
+        test_endpoint,
+        headers=service_headers,
+        json=test_inputs)
+
+    print("Portability service response:\n{}".format(
+        json.dumps(json.loads(response.text), indent=4)), flush=True)
+
+    test_id = json.loads(response.text)["test_id"]
+
+    return test_id
+
+def print_test_states(test_states):
+    """Print the states of tests."""
+    print("{:%Y-%m-%dT%H:%M:%SZ}".format(datetime.datetime.utcnow()), flush=True)
```

Comment:
Did you find that you needed to use `flush=True` to get it to properly write to stdout? I've never found I needed to use this.

Reply:
I had to do that for travis or else it would kill the job after 10 minutes of what looked like inactivity.

Reply:
Interesting. How does flush produce "activity"? Something weird about how travis interacts with the stdout stream? (not important, just interested)

Reply:
Yeah, travis might be interacting with stdout differently. This is the test that was fixed by adding `flush=True`: https://travis-ci.org/HumanCellAtlas/skylab/builds/326686167.

Adding flush changed it to this: https://travis-ci.org/HumanCellAtlas/skylab/builds/326688523

Reply:
Weird! ok, thanks for showing.